tests/b2share_functional_tests/test_schema_cli.py

```diff
@@ -0,0 +1,97 @@
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Test B2Share schema cli module."""
+import json
+import pytest
+from click.testing import CliRunner
+from flask_cli import ScriptInfo
+
+from b2share.modules.communities.api import Community
+from b2share.modules.communities.cli import communities as communities_cmd
+from b2share.modules.schemas.cli import schemas as schemas_cmd
+
+test_schema = {
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "EISCAT Metadata",
+    "description": "This is the blueprint of the metadata block specific for a unit test community",
+    "properties": {
+        "some_field": {
+            "description": "The only field in this schema",
+            "title": "Field nr 1",
+            "type": "string",
+        }
+    },
+    "type": "object",
+    "additionalProperties": False,
+}
+
+configurations = [({'config': {'PREFERRED_URL_SCHEME': 'https'}}),
+                  ({'config': {'PREFERRED_URL_SCHEME': 'http'}})]
+
+@pytest.mark.parametrize('app', configurations, indirect=['app'])
+def test_existing_community_set_schema_cmd(app, test_communities):
+    """Test the `schemas set_schema` CLI command."""
+    with app.app_context():
+        runner = CliRunner()
+        script_info = ScriptInfo(create_app=lambda info: app)
+        comm_name = test_communities.popitem()[0]
+        with runner.isolated_filesystem():
+            with open("schema.json", "w") as f:
+                f.write(json.dumps(test_schema))
+
+            # Run 'set schema' command
+            args = ['set_schema', comm_name, 'schema.json']
+            result = runner.invoke(communities_cmd, args, obj=script_info)
+            assert result.exit_code == 0
+
+            # Run 'set schema' command again
+            with open("schema.json", "w") as f:
+                f.write(json.dumps(test_schema))
+            args = ['set_schema', comm_name, 'schema.json']
+            result = runner.invoke(communities_cmd, args, obj=script_info)
+            assert result.exit_code == 0
+
+
+
+@pytest.mark.parametrize('app', configurations, indirect=['app'])
+def test_new_community_set_schema_cmd(app):
+    """Test adding a community and setting its schema using CLI commands."""
+    with app.app_context():
+        runner = CliRunner()
+        script_info = ScriptInfo(create_app=lambda info: app)
+        comm_name = 'MyCom'
+
+        # Create a new community on the command line
+        args = ['create', comm_name, 'MyCom description', 'eudat.png']
+        result = runner.invoke(communities_cmd, args, obj=script_info)
+        assert result.exit_code == 0
+
+        with runner.isolated_filesystem():
+            with open("schema.json", "w") as f:
+                f.write(json.dumps(test_schema))
+
+            # initialize the root schema in the test environment
+            result = runner.invoke(schemas_cmd, ['init'], obj=script_info)
+            assert result.exit_code == 0
```

Comment:
Ok for now, but normally the test should also check that the command actually did something. It could return 0 by mistake, not commit at the end or have another issue.